src/services/post.ts

```diff
@@ -29,13 +24,11 @@ class PostsService {
 
     await post.save();
 
-    return { message: 'Post have been created' };
+    return;
   }
 
   async getAllPosts() {
-    const posts = await Post.find().populate('user').exec();
-
-    return posts;
+    return await Post.find().lean().populate('user').exec();
```

Comment:
```suggestion
    return await Post.find().populate('user').lean();
```

If you want to populate only certain fields you can do such:
```
    return await Post.find().populate('user', 'userName').lean();
```